sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusMessageSerializer.java

```diff
@@ -143,34 +140,84 @@ public int getSize(Message amqpMessage) {
         if (amqpMessage.getProperties() == null) {
             amqpMessage.setProperties(new Properties());
         }
-
         amqpMessage.setMessageId(brokeredMessage.getMessageId());
         amqpMessage.setContentType(brokeredMessage.getContentType());
         amqpMessage.setCorrelationId(brokeredMessage.getCorrelationId());
-        amqpMessage.setSubject(brokeredMessage.getLabel());
-        amqpMessage.getProperties().setTo(brokeredMessage.getTo());
+        amqpMessage.setSubject(brokeredMessage.getSubject());
         amqpMessage.setReplyTo(brokeredMessage.getReplyTo());
         amqpMessage.setReplyToGroupId(brokeredMessage.getReplyToSessionId());
         amqpMessage.setGroupId(brokeredMessage.getSessionId());
 
+        final AmqpMessageProperties brokeredProperties = brokeredMessage.getAmqpAnnotatedMessage().getProperties();
+
+        amqpMessage.setContentEncoding(brokeredProperties.getContentEncoding());
+        if (brokeredProperties.getGroupSequence() != null) {
+            amqpMessage.setGroupSequence(brokeredProperties.getGroupSequence());
+        }
+        amqpMessage.getProperties().setTo(brokeredMessage.getTo());
+        amqpMessage.getProperties().setUserId(new Binary(brokeredProperties.getUserId()));
+
+        if (brokeredProperties.getAbsoluteExpiryTime() != null) {
+            amqpMessage.getProperties().setAbsoluteExpiryTime(Date.from(brokeredProperties.getAbsoluteExpiryTime().
+                toInstant()));
+        }
+        if (brokeredProperties.getCreationTime() != null) {
+            amqpMessage.getProperties().setCreationTime(Date.from(brokeredProperties.getCreationTime().toInstant()));
+        }
+
+        //set footer
+        amqpMessage.setFooter(new Footer(brokeredMessage.getAmqpAnnotatedMessage().getFooter()));
+
+        //set header
+        AmqpMessageHeader header = brokeredMessage.getAmqpAnnotatedMessage().getHeader();
+        if (header.getDeliveryCount() != null) {
+            amqpMessage.setDeliveryCount(brokeredMessage.getAmqpAnnotatedMessage().getHeader().getDeliveryCount());
```

Comment:
You already have a local reference to the `header`, you can use that and go `header.getDeliveryCount()`. Same with other instances.